agent/functional_tests/tests/functionaltests_windows_test.go

```diff
@@ -102,14 +98,19 @@ func TestAwslogsDriver(t *testing.T) {
 		LogGroupName:  aws.String(awslogsLogGroupName),
 		LogStreamName: aws.String(fmt.Sprintf("ecs-functional-tests/awslogs/%s", taskId)),
 	}
-	resp, err := cwlClient.GetLogEvents(params)
-	if err != nil {
-		t.Fatalf("CloudWatchLogs get log failed: %v", err)
-	}
 
-	if len(resp.Events) != 1 {
-		t.Errorf("Get number of log events: %d", len(resp.Events))
+	var resp *cloudwatchlogs.GetLogEventsOutput
+	// The test could fail for timing issue, so retry for 30 seconds to make this test more stable
+	for i := 0; i < 30; i++ {
+		resp, err = cwlClient.GetLogEvents(params)
```

Comment:
Looks like the `retryer` is only evaluated when the request failed per the code [here](https://github.com/aws/aws-sdk-go/blob/v1.0.0/aws/request/request.go#L205) which isn't the case here. Any better suggestion, otherwise I'm going to use `RetryWithBackoff` [here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/utils/utils.go#L110)?